plugins/inputs/x509_cert/x509_cert_test.go

```diff
@@ -260,6 +262,36 @@ func TestGatherChain(t *testing.T) {
 	}
 }
 
+func TestGatherUDPCert(t *testing.T) {
+	pair, err := tls.X509KeyPair([]byte(pki.ReadServerCert()), []byte(pki.ReadServerKey()))
+	require.NoError(t, err)
+
+	cfg := &dtls.Config{
+		Certificates: []tls.Certificate{pair},
+	}
+
+	addr := &net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: 0}
+	listener, err := dtls.Listen("udp", addr, cfg)
+	require.NoError(t, err)
+	defer listener.Close()
+
+	go func() {
+		_, err := listener.Accept()
+		require.NoError(t, err)
+	}()
+
+	m := &X509Cert{
+		Sources: []string{"udp://" + listener.Addr().String()},
+	}
```

Comment:
You need to set the logging facility here or you will get segfaults (that are the failing CI tests you see):
```suggestion
	m := &X509Cert{
		Sources: []string{"udp://" + listener.Addr().String()},
		Log:     testutil.Logger{},
	}
```